### ISSUE TITLE
feat: implement `capture`

### ISSUE THREAD

**Issue by okwasniewski** [callstack/ts-regex#16]
## Describe the feature

Implement `capture()` component

```ts
let regex = buildRegex(
    "a", capture(one("c"), oneOrMore("ab")) // a(c(?:ab)+)
)
```

## References

https://developer.apple.com/documentation/regexbuilder/capture

**Comment by mdjastrzebski** [comment]
My suggestions:

Swift:
```swift
Regex {
    // Basic
    Capture {
      OneOrMore(.digit)
      "."
      Repeat(.digit, count: 2)
    }
   // With transform
    Capture {
      OneOrMore(.digit)
      "."
      Repeat(.digit, count: 2)
    } transform: { Double($0)! }
}
```


Proposed API:
```ts
buildRegex(
  // Basic
  capture(
     oneOrMore(digit),
     ".",
     repeat({ count: 2}, digit),
  )
  // With transform
  capture(
     { transform: (n) => parseFloat(n) },
     oneOrMore(digit),
     ".",
     repeat({ count: 2}, digit),
  )
}
```

The goal would be to generate a basic capture group (...) around the passed list of elements. Providing transform is optional.

I don't know if config (transform) should be the first or last parameter. JS conventions would suggest first due to the presence of vargargs.
Note that we probably should keep the transform key explicit, as the Swift API also has some as an argument, which we might want to implement)

**Comment by mdjastrzebski** [comment]
Implemented in #24
